pkg/cluster/cluster.go

```diff
@@ -1034,10 +1034,15 @@ func (c *Cluster) processPodEvent(obj interface{}) error {
 
 	c.podSubscribersMu.RLock()
 	subscriber, ok := c.podSubscribers[spec.NamespacedName(event.PodName)]
-	c.podSubscribersMu.RUnlock()
 	if ok {
-		subscriber <- event
+		select {
+		case subscriber <- event:
+		default:
+			// we end up here when there is no receiver of the channel
+			// avoiding a deadlock: https://gobyexample.com/non-blocking-channel-operations
+		}
 	}
+	c.podSubscribersMu.RUnlock()
```

Comment:
What is the reason to protect channel operations by an unrealted mutex ?

Reply:
podSubscribersMu will also be locked when closing the channel in c.unregisterPodSubscriber. This should prevent the panic when writing to a closed channel